src/lgar.cxx

```diff
@@ -1587,11 +1643,15 @@ extern void lgar_wetting_fronts_cross_layer_boundary(int num_layers,
       current->dzdt_cm_per_h = 0;
       current->to_bottom = TRUE;
       next->to_bottom = FALSE;
+
+      if (isinf(next->depth_cm)){//in some rare cases, due to (very) different shapes of soil water rentention curves in adjacent soil layers near saturation, the WF that crossed a layer bdy can yield a theta value identical to the one below it, resulting in division by 0 and an infinite WF depth. 
```

Comment:
the code should never assign infinity to a wetting front depth, instead of checking it here, the code should handle it early on. I think the right place is:

`double mbal_Z_correction = mbal_correction / (theta_new - next_to_next->theta);`

We should also add an assertion
`assert (wetting_front->depth_cm > domain_depth)`

Reply:
Let's chat about this. I updated the code as you suggested, fixing the error at mbal_Z_correction rather than when we assign the WF depth, but I don't think we should add that assertion. This is because we basically have a whole function for crossing the domain lower boundary that handles that case.

Reply:
okay, so I guess this was the only place where WF depth was crossing the domain boundary depth, correct?

Reply:
Well, generally WF depth can cross the domain boundary depth any time WF depth is updated, so this is not the only place. However, the cases of wetting fronts crossing a layer boundary, wetting fronts crossing a domain boundary, and wetting fronts merging with one another are handled specially by an individual function for each of those, and in specific places. The code you commented on here is in the function where WFs cross a layer boundary, so in this function we don't deal with the case where a WF crosses the domain bdy.

Reply:
I should add, this is probably the only place where WF depth can be calculated as infinite. This is because this is probably the only place where we have a delta theta in the denominator used to update WF depth (aside from in dzdt function, where we also deal with the issue of 0 in the denominator).